flaml/automl.py

```diff
@@ -1085,14 +1128,25 @@ def custom_metric(
                 hyperparamter configurations for the corresponding estimators.
             seed: int or None, default=None | The random seed for np.random.
             **fit_kwargs: Other key word arguments to pass to fit() function of
-                the searched learners, such as sample_weight.
+                the searched learners, such as sample_weight. Include period and
+                freq as key word arguments for forecast task.
         '''
         self._start_time_flag = time.time()
         self._state.task = task
         self._state.log_training_metric = log_training_metric
         self._state.fit_kwargs = fit_kwargs
         self._state.weight_val = sample_weight_val
         self._state.groups = groups
+
+        if self._state.task == 'forecast':
+            if dataframe is not None:
+                if ('ds' not in dataframe) or ('y' not in dataframe):
+                    raise ValueError(
+                        'Dataframe must have columns "ds" and "y" with the dates and '
+                        'values respectively.'

```

Comment:
This error message can be made more informative such as asking if user's intent is to perform forecasting, saying that for forecasting task currently requires the dataframe to contains specific columns for the timestamps and labels.

Reply:
changes made locally